### PR TITLE
[RHOSINFRA-92] Avoid infinite references to the lookup values

### PR DESCRIPTION
**Pull request by obaranov**
Fixed issue with the infinite lookup

@aopincar, @bregman-arie please review.

### CODE REVIEW

**Comment by obaranov** [comment]
fixing unit tests

**Comment by yfried-redhat** [comment]
@aopincar please review this

**Comment by yfried-redhat** [comment]
:+1: LGTM
@aopincar please review and merge

File: tests/test_cwd/lookup_new_style_circular_reference.yml [hunk 1]
```diff
@@ -0,0 +1,5 @@
+foo:
+  test1: "{{ !lookup bar.test1 }}"
+
+bar:
+  test1: "{{ !lookup foo.test1 }}"
```
Comment (aopincar) [inline tests/test_cwd/lookup_new_style_circular_reference.yml]: Please add new line at the end of each file

File: cli/yamls.py [hunk 2]
```diff
@@ -141,12 +141,29 @@ def replace_lookup(self):
...
+
+                    lookup_key = lookup_key.group(0).strip()[:-2].strip()
+                    # check if lookups are in circular reference
+                    if lookup_key in visited_lookups:
+                        raise exceptions.IRInfiniteLookupException(
+                            visited_lookups)
+                    visited_lookups.append(lookup_key)
+                    lookup_value = self.dict_lookup(lookup_key.split("."))
+
+                    if not isinstance(lookup_value, Lookup):
```
Comment (aopincar) [inline cli/yamls.py]: The entire 'if' block is not needed.
We're no longer support old style lookups ('!lookup key.subkey') only new style lookups ('{{!lookup key.subkey}}') so there will be no Look objects...

**Comment by obaranov** [comment]
retest

**Comment by yfried-redhat** [comment]
retest

**Pull request merged**